<?xml version="1.0"?>
<launch>

  <!-- Sample launch file, transforms NED referenced data in imu_link_ned to ENU equivalent data in imu_link -->

  <!-- Static tfs -->
  <node pkg="tf" type="static_transform_publisher" name="tf_imu_ned_enu_global"
        args="0 0 0 1.5708 0 3.1416 imu_link imu_link_ned_global 10" output="screen"/>
  <node pkg="tf" type="static_transform_publisher" name="tf_imu_ned_enu_local"
        args="0 0 0 0 0 3.1416 imu_link imu_link_ned_local 10" output="screen"/>

  <!-- Transforms NED IMU data pulished in frame imu_link_ned on topics /imu/data_ned and /imu/mag_ned to
   ENU data in frame imu_link published on /imu/data_enu and /imu/mag_enu -->
  <node pkg="imu_transformer" type="imu_transformer_node" name="imu_data_transformer" output="screen">
    <remap from="imu_in" to="imu_ned"/>
    <remap from="imu_out" to="imu_enu"/>
    <param name="target_frame" value="imu_link"/>
    <rosparam>
      source_frame:
        global: imu_link_ned_global
        local: imu_link_ned_local
    </rosparam>
  </node>

</launch>
